### DISCUSSION TITLE
List[CustomClass] does not work in FastAPI inside Pydantic model

### DISCUSSION THREAD

**Discussion by yavuzakyazici** [fastapi/fastapi#10681] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class User(BaseModel):
    UserId: int
    FirstName: str
    LastName: str
    Phone: Optional[str]
    Email: str
    UserLogin: str
    UserPassword: str
    IsActive: bool
    Settings: Optional[str]
    ProfileImgUrl: Optional[str]
    LockedLessons: Optional[str]
    CompletedLessons: Optional[str]
    UserScore: int = 0
    RegisterDate: datetime = None
    UpdateDate: datetime = None

    Addresses: list[Address] | None = None

    class Config:
        from_attributes = True
        arbitrary_types_allowed = True


class Address(BaseModel):
    AddressId: int
    UserId: int
    Addr1: str
    Addr2: Optional[str]
    District: Optional[str]
    City: str
    St: Optional[str]
    Country: str
    Zip: Optional[str]

    class Config:
        from_attributes = True
```

### Description

I can't seem to understand not being able to access another class on the same page as all my pydantic models. VsCode says "Address" is not definedPylance[reportUndefinedVariable] https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUndefinedVariable

### Operating System

macOS

### Operating System Details

I am running MacOS Sonoma

### FastAPI Version

0.104.1

### Pydantic Version

2.5.1

### Python Version

 3.12.0

### Additional Context

<img width="840" alt="Screenshot 2023-11-19 at 17 52 12" src="https://github.com/tiangolo/fastapi/assets/148442912/355bab4d-7c8e-43ee-bff2-61e89c90e4b0">

**Comment by sriram-kondakindi** [comment, marked as answer]
Define address class on top of User class

**Comment by yavuzakyazici** [comment]
Yes, it worked :))))) ..  I guess scope also has to do with thinking about the code is executing from the firs line up all the way down to the end ...  Thanks a lot for that :))

**Comment by hasansezertasan** [comment]
```python
from pydantic import BaseModel
from typing import Optional


class User(BaseModel):
    username: str
    addresses: list["Address"] | None = None

class Address(BaseModel):
    zip: Optional[str]

```

You can also do it like this. I'm not sure which Python versions support this quotation feature but "re-ordering" the code just for this doesn't feel right.
